Fixed docstring errors in torch/nn/functional.py (Docathon H2)

pilot-j opened the pull request:
Fixes #112597 
### Output:
**BEFORE:**
```functional.py:1 at module level:
        D400: First line should end with a period (not 'e')
functional.py:438 in public function `fractional_max_pool2d_with_indices`:
        D400: First line should end with a period (not ')')
functional.py:537 in public function `fractional_max_pool3d_with_indices`:
        D400: First line should end with a period (not ')')
functional.py:646 in public function `max_pool1d_with_indices`:
        D400: First line should end with a period (not ')')
functional.py:732 in public function `max_pool2d_with_indices`:
        D400: First line should end with a period (not ')')
functional.py:818 in public function `max_pool3d_with_indices`:
        D400: First line should end with a period (not ')')
functional.py:932 in public function `max_unpool1d`:
        D401: First line should be in imperative mood (perhaps 'Compute', not 'Computes')
functional.py:968 in public function `max_unpool2d`:
        D401: First line should be in imperative mood (perhaps 'Compute', not 'Computes')
functional.py:1000 in public function `max_unpool3d`:
        D401: First line should be in imperative mood (perhaps 'Compute', not 'Computes')
functional.py:1031 in public function `lp_pool2d`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:1031 in public function `lp_pool2d`:
        D400: First line should end with a period (not 'f')
functional.py:1031 in public function `lp_pool2d`:
        D401: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
functional.py:1056 in public function `lp_pool1d`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:1056 in public function `lp_pool1d`:
        D400: First line should end with a period (not 'f')
functional.py:1056 in public function `lp_pool1d`:
        D401: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
functional.py:1077 in public function `adaptive_max_pool1d_with_indices`:
        D400: First line should end with a period (not ')')
functional.py:1119 in public function `adaptive_max_pool2d_with_indices`:
        D400: First line should end with a period (not ')')
functional.py:1163 in public function `adaptive_max_pool3d_with_indices`:
        D400: First line should end with a period (not ')')
functional.py:1220 in public function `adaptive_avg_pool2d`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:1220 in public function `adaptive_avg_pool2d`:
        D400: First line should end with a period (not 'f')
functional.py:1220 in public function `adaptive_avg_pool2d`:
        D401: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
functional.py:1237 in public function `adaptive_avg_pool3d`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:1237 in public function `adaptive_avg_pool3d`:
        D400: First line should end with a period (not 'f')
functional.py:1237 in public function `adaptive_avg_pool3d`:
        D401: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
functional.py:1255 in public function `dropout`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:1255 in public function `dropout`:
        D400: First line should end with a period (not 't')
functional.py:1275 in public function `alpha_dropout`:
        D401: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
functional.py:1287 in public function `dropout1d`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:1287 in public function `dropout1d`:
        D400: First line should end with a period (not ',')
functional.py:1325 in public function `dropout2d`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:1325 in public function `dropout2d`:
        D400: First line should end with a period (not ',')
functional.py:1369 in public function `dropout3d`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:1369 in public function `dropout3d`:
        D400: First line should end with a period (not ',')
functional.py:1408 in public function `feature_alpha_dropout`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:1408 in public function `feature_alpha_dropout`:
        D400: First line should end with a period (not ',')
functional.py:1466 in public function `relu`:
        D400: First line should end with a period (not 'r')
functional.py:1466 in public function `relu`:
        D402: First line should not be the function's "signature"
functional.py:1491 in public function `glu`:
        D400: First line should end with a period (not 'r')
functional.py:1491 in public function `glu`:
        D402: First line should not be the function's "signature"
functional.py:1516 in public function `hardtanh`:
        D400: First line should end with a period (not 'r')
functional.py:1516 in public function `hardtanh`:
        D402: First line should not be the function's "signature"
functional.py:1542 in public function `relu6`:
        D400: First line should end with a period (not 'r')
functional.py:1542 in public function `relu6`:
        D402: First line should not be the function's "signature"
functional.py:1558 in public function `elu`:
        D401: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
functional.py:1582 in public function `selu`:
        D400: First line should end with a period (not 'r')
functional.py:1582 in public function `selu`:
        D402: First line should not be the function's "signature"
functional.py:1611 in public function `celu`:
        D400: First line should end with a period (not 'r')
functional.py:1611 in public function `celu`:
        D402: First line should not be the function's "signature"
functional.py:1638 in public function `leaky_relu`:
        D400: First line should end with a period (not 'r')
functional.py:1638 in public function `leaky_relu`:
        D402: First line should not be the function's "signature"
functional.py:1688 in public function `rrelu`:
        D400: First line should end with a period (not 'r')
functional.py:1688 in public function `rrelu`:
        D402: First line should not be the function's "signature"
functional.py:1755 in public function `tanhshrink`:
        D400: First line should end with a period (not 'r')
functional.py:1755 in public function `tanhshrink`:
        D402: First line should not be the function's "signature"
functional.py:1767 in public function `softsign`:
        D400: First line should end with a period (not 'r')
functional.py:1767 in public function `softsign`:
        D402: First line should not be the function's "signature"
functional.py:1806 in public function `softmin`:
        D401: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
functional.py:1832 in public function `softmax`:
        D401: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
functional.py:1868 in public function `gumbel_softmax`:
        D401: First line should be in imperative mood (perhaps 'Sample', not 'Samples')
functional.py:1930 in public function `log_softmax`:
        D401: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
functional.py:1969 in public function `tanh`:
        D400: First line should end with a period (not 'r')
functional.py:1969 in public function `tanh`:
        D402: First line should not be the function's "signature"
functional.py:1980 in public function `sigmoid`:
        D400: First line should end with a period (not 'r')
functional.py:1980 in public function `sigmoid`:
        D402: First line should not be the function's "signature"
functional.py:1990 in public function `hardsigmoid`:
        D400: First line should end with a period (not 'n')
functional.py:1990 in public function `hardsigmoid`:
        D401: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
functional.py:2057 in public function `silu`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:2057 in public function `silu`:
        D401: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
functional.py:2081 in public function `mish`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:2081 in public function `mish`:
        D401: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
functional.py:2100 in public function `hardswish`:
        D400: First line should end with a period (not ':')
functional.py:2100 in public function `hardswish`:
        D401: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
functional.py:2136 in public function `embedding`:
        D202: No blank lines allowed after function docstring (found 1)
functional.py:2136 in public function `embedding`:
        D401: First line should be in imperative mood; try rephrasing (found 'A')
functional.py:2254 in public function `embedding_bag`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:2254 in public function `embedding_bag`:
        D400: First line should end with a period (not 'e')
functional.py:2254 in public function `embedding_bag`:
        D401: First line should be in imperative mood (perhaps 'Compute', not 'Computes')
functional.py:2462 in public function `batch_norm`:
        D401: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
functional.py:2507 in public function `instance_norm`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:2507 in public function `instance_norm`:
        D400: First line should end with a period (not 'a')
functional.py:2507 in public function `instance_norm`:
        D401: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
functional.py:2540 in public function `layer_norm`:
        D401: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
functional.py:2554 in public function `group_norm`:
        D401: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
functional.py:2567 in public function `local_response_norm`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:2567 in public function `local_response_norm`:
        D400: First line should end with a period (not 'f')
functional.py:2567 in public function `local_response_norm`:
        D401: First line should be in imperative mood (perhaps 'Apply', not 'Applies')
functional.py:2611 in public function `ctc_loss`:
        D401: First line should be in imperative mood; try rephrasing (found 'The')
functional.py:2679 in public function `nll_loss`:
        D401: First line should be in imperative mood; try rephrasing (found 'The')
functional.py:2895 in public function `kl_div`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:2895 in public function `kl_div`:
        D400: First line should end with a period (not 's')
functional.py:2895 in public function `kl_div`:
        D401: First line should be in imperative mood; try rephrasing (found 'The')
functional.py:2978 in public function `cross_entropy`:
        D401: First line should be in imperative mood; try rephrasing (found 'This')
functional.py:3069 in public function `binary_cross_entropy`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:3069 in public function `binary_cross_entropy`:
        D400: First line should end with a period (not 't')
functional.py:3069 in public function `binary_cross_entropy`:
        D401: First line should be in imperative mood; try rephrasing (found 'Function')
functional.py:3139 in public function `binary_cross_entropy_with_logits`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:3139 in public function `binary_cross_entropy_with_logits`:
        D400: First line should end with a period (not 't')
functional.py:3139 in public function `binary_cross_entropy_with_logits`:
        D401: First line should be in imperative mood; try rephrasing (found 'Function')
functional.py:3211 in public function `smooth_l1_loss`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:3211 in public function `smooth_l1_loss`:
        D400: First line should end with a period (not 'e')
functional.py:3211 in public function `smooth_l1_loss`:
        D401: First line should be in imperative mood; try rephrasing (found 'Function')
functional.py:3251 in public function `huber_loss`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:3251 in public function `huber_loss`:
        D400: First line should end with a period (not 'e')
functional.py:3251 in public function `huber_loss`:
        D401: First line should be in imperative mood; try rephrasing (found 'Function')
functional.py:3282 in public function `l1_loss`:
        D400: First line should end with a period (not 'r')
functional.py:3282 in public function `l1_loss`:
        D402: First line should not be the function's "signature"
functional.py:3313 in public function `mse_loss`:
        D400: First line should end with a period (not 'r')
functional.py:3313 in public function `mse_loss`:
        D402: First line should not be the function's "signature"
functional.py:3346 in public function `margin_ranking_loss`:
        D400: First line should end with a period (not 'r')
functional.py:3346 in public function `margin_ranking_loss`:
        D402: First line should not be the function's "signature"
functional.py:3382 in public function `hinge_embedding_loss`:
        D400: First line should end with a period (not 'r')
functional.py:3382 in public function `hinge_embedding_loss`:
        D402: First line should not be the function's "signature"
functional.py:3411 in public function `multilabel_margin_loss`:
        D400: First line should end with a period (not 'r')
functional.py:3411 in public function `multilabel_margin_loss`:
        D402: First line should not be the function's "signature"
functional.py:3439 in public function `soft_margin_loss`:
        D400: First line should end with a period (not 'r')
functional.py:3439 in public function `soft_margin_loss`:
        D402: First line should not be the function's "signature"
functional.py:3462 in public function `multilabel_soft_margin_loss`:
        D400: First line should end with a period (not 'r')
functional.py:3462 in public function `multilabel_soft_margin_loss`:
        D402: First line should not be the function's "signature"
functional.py:3510 in public function `cosine_embedding_loss`:
        D400: First line should end with a period (not 'r')
functional.py:3510 in public function `cosine_embedding_loss`:
        D402: First line should not be the function's "signature"
functional.py:3543 in public function `multi_margin_loss`:
        D400: First line should end with a period (not 'r')
functional.py:3543 in public function `multi_margin_loss`:
        D402: First line should not be the function's "signature"
functional.py:3708 in public function `upsample` (skipping F811,B950):
        D103: Missing docstring in public function
functional.py:3713 in public function `upsample` (skipping F811,B950):
        D103: Missing docstring in public function
functional.py:3718 in public function `upsample` (skipping F811):
        D205: 1 blank line required between summary line and description (found 0)
functional.py:3718 in public function `upsample` (skipping F811):
        D400: First line should end with a period (not 'n')
functional.py:3783 in private function `_is_integer`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:3794 in public function `interpolate` (skipping F811,B950):
        D103: Missing docstring in public function
functional.py:3799 in public function `interpolate` (skipping F811,B950):
        D103: Missing docstring in public function
functional.py:3804 in public function `interpolate` (skipping F811,B950):
        D103: Missing docstring in public function
functional.py:3809 in public function `interpolate` (skipping F811):
        D103: Missing docstring in public function
functional.py:3821 in public function `interpolate` (skipping F811,B950):
        D205: 1 blank line required between summary line and description (found 0)
functional.py:3821 in public function `interpolate` (skipping F811,B950):
        D400: First line should end with a period (not 'n')
functional.py:4062 in public function `upsample_nearest` (skipping F811):
        D103: Missing docstring in public function
functional.py:4067 in public function `upsample_nearest` (skipping F811):
        D103: Missing docstring in public function
functional.py:4100 in public function `upsample_bilinear` (skipping F811):
        D103: Missing docstring in public function
functional.py:4107 in public function `upsample_bilinear` (skipping F811):
        D103: Missing docstring in public function
functional.py:4114 in public function `upsample_bilinear` (skipping F811):
        D103: Missing docstring in public function
functional.py:4121 in public function `upsample_bilinear` (skipping F811):
        D103: Missing docstring in public function
functional.py:4174 in public function `grid_sample`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:4174 in public function `grid_sample`:
        D400: First line should end with a period (not 'e')
functional.py:4315 in public function `affine_grid`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:4315 in public function `affine_grid`:
        D400: First line should end with a period (not 'f')
functional.py:4315 in public function `affine_grid`:
        D401: First line should be in imperative mood (perhaps 'Generate', not 'Generates')
functional.py:4608 in public function `triplet_margin_loss`:
        D200: One-line docstring should fit on one line with quotes (found 3)
functional.py:4608 in public function `triplet_margin_loss`:
        D400: First line should end with a period (not 's')
functional.py:4643 in public function `triplet_margin_with_distance_loss`:
        D200: One-line docstring should fit on one line with quotes (found 3)
functional.py:4705 in public function `normalize`:
        D401: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
functional.py:4733 in public function `assert_int_or_pair`:
        D103: Missing docstring in public function
functional.py:4743 in public function `unfold`:
        D401: First line should be in imperative mood (perhaps 'Extract', not 'Extracts')
functional.py:4773 in public function `fold`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:4773 in public function `fold`:
        D400: First line should end with a period (not 'g')
functional.py:4773 in public function `fold`:
        D401: First line should be in imperative mood (perhaps 'Combine', not 'Combines')
functional.py:4800 in private function `_in_projection_packed`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:4800 in private function `_in_projection_packed`:
        D401: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
functional.py:4867 in private function `_in_projection`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:4867 in private function `_in_projection`:
        D400: First line should end with a period (not 'y')
functional.py:4867 in private function `_in_projection`:
        D401: First line should be in imperative mood (perhaps 'Perform', not 'Performs')
functional.py:5128 in public function `multi_head_attention_forward`:
        D205: 1 blank line required between summary line and description (found 0)
functional.py:5128 in public function `multi_head_attention_forward`:
        D400: First line should end with a period (not ':')
160
```

**AFTER:**

```
functional.py:3709 in public function `upsample` (skipping F811,B950):
        D103: Missing docstring in public function
functional.py:3714 in public function `upsample` (skipping F811,B950):
        D103: Missing docstring in public function
functional.py:3798 in public function `interpolate` (skipping F811,B950):
        D103: Missing docstring in public function
functional.py:3803 in public function `interpolate` (skipping F811,B950):
        D103: Missing docstring in public function
functional.py:3808 in public function `interpolate` (skipping F811,B950):
        D103: Missing docstring in public function
functional.py:3813 in public function `interpolate` (skipping F811):
        D103: Missing docstring in public function
functional.py:4068 in public function `upsample_nearest` (skipping F811):
        D103: Missing docstring in public function
functional.py:4073 in public function `upsample_nearest` (skipping F811):
        D103: Missing docstring in public function
functional.py:4106 in public function `upsample_bilinear` (skipping F811):
        D103: Missing docstring in public function
functional.py:4113 in public function `upsample_bilinear` (skipping F811):
        D103: Missing docstring in public function
functional.py:4120 in public function `upsample_bilinear` (skipping F811):
        D103: Missing docstring in public function
functional.py:4127 in public function `upsample_bilinear` (skipping F811):
        D103: Missing docstring in public function
functional.py:4742 in public function `assert_int_or_pair`:
        D103: Missing docstring in public function
13
``` 

The file contained several docstring errors. I have fixed all of them(hopefully) and have tried to improve the over all readability of the code. For most part, I have included relevant description of functions (referred from official PyTorch Docs). In some cases where functions are purely mathematical or it is difficult to give one line description, I have just included references.

For testing, I relied on local system and created a separate file. For final edits, I directly changed the contents of forked repo as visible already.

Kindly review @svekars @subramen @kit1980 



cc @svekars @carljparker

pilot-j committed:
doctring errors corrected till 1490

pilot-j committed:
docstring changes till dropout (1254)

pilot-j committed:
Update functional.py

pilot-j committed:
Update functional.py

pilot-j committed:
48 remaining - margin_loss section

pilot-j committed:
Fixed all docstring errors

svekars commented on torch/nn/functional.py:
```diff
@@ -1284,9 +1282,9 @@ def alpha_dropout(input: Tensor, p: float = 0.5, training: bool = False, inplace
-    r"""
-    Randomly zero out entire channels (a channel is a 1D feature map,
-    e.g., the :math:`j`-th channel of the :math:`i`-th sample in the
+    r"""Randomly zero out entire channels (a channel is a 1D feature map.
```
```suggestion
    r"""Randomly zero out entire channels a channel is a 1D feature map.
```

svekars commented on torch/nn/functional.py:
```diff
@@ -1284,9 +1282,9 @@ def alpha_dropout(input: Tensor, p: float = 0.5, training: bool = False, inplace
-    e.g., the :math:`j`-th channel of the :math:`i`-th sample in the
+    r"""Randomly zero out entire channels (a channel is a 1D feature map.
+    
+    E.g., the :math:`j`-th channel of the :math:`i`-th sample in the
```
```suggestion
    For example, the :math:`j`-th channel of the :math:`i`-th sample in the
```

svekars commented on torch/nn/functional.py:
```diff
@@ -1322,9 +1320,9 @@ def dropout1d(input: Tensor, p: float = 0.5, training: bool = True, inplace: boo
-    r"""
-    Randomly zero out entire channels (a channel is a 2D feature map,
-    e.g., the :math:`j`-th channel of the :math:`i`-th sample in the
+    r"""Randomly zero out entire channels (a channel is a 2D feature map.
```
```suggestion
    r"""Randomly zero out entire channels a channel is a 2D feature map.
```

svekars commented on torch/nn/functional.py:
```diff
@@ -1322,9 +1320,9 @@ def dropout1d(input: Tensor, p: float = 0.5, training: bool = True, inplace: boo
-    e.g., the :math:`j`-th channel of the :math:`i`-th sample in the
+    r"""Randomly zero out entire channels (a channel is a 2D feature map.
+    
+    E.g., the :math:`j`-th channel of the :math:`i`-th sample in the
```
```suggestion
    For example, the :math:`j`-th channel of the :math:`i`-th sample in the
```

svekars commented on torch/nn/functional.py:
```diff
@@ -1322,9 +1320,9 @@ def dropout1d(input: Tensor, p: float = 0.5, training: bool = True, inplace: boo
+    r"""Randomly zero out entire channels (a channel is a 2D feature map.
+    
+    E.g., the :math:`j`-th channel of the :math:`i`-th sample in the
     batched input is a 2D tensor :math:`\text{input}[i, j]`) of the input tensor).
```
```suggestion
    batched input is a 2D tensor :math:`\text{input}[i, j]`) of the input tensor.
```

svekars commented on torch/nn/functional.py:
```diff
@@ -1366,9 +1364,9 @@ def dropout2d(input: Tensor, p: float = 0.5, training: bool = True, inplace: boo
-    r"""
-    Randomly zero out entire channels (a channel is a 3D feature map,
-    e.g., the :math:`j`-th channel of the :math:`i`-th sample in the
+    r"""Randomly zero out entire channels (a channel is a 3D feature map.
```
```suggestion
    r"""Randomly zero out entire channels a channel is a 3D feature map.
```

svekars commented on torch/nn/functional.py:
```diff
@@ -1366,9 +1364,9 @@ def dropout2d(input: Tensor, p: float = 0.5, training: bool = True, inplace: boo
-    e.g., the :math:`j`-th channel of the :math:`i`-th sample in the
+    r"""Randomly zero out entire channels (a channel is a 3D feature map.
+    
+    E.g., the :math:`j`-th channel of the :math:`i`-th sample in the
     batched input is a 3D tensor :math:`\text{input}[i, j]`) of the input tensor).
```
```suggestion
    For example, the :math:`j`-th channel of the :math:`i`-th sample in the
    batched input is a 3D tensor :math:`\text{input}[i, j]`) of the input tensor.
```

svekars commented on torch/nn/functional.py:
```diff
@@ -1764,9 +1757,9 @@ def tanhshrink(input):
 
 def softsign(input):
-    r"""softsign(input) -> Tensor
+    r"""Apply the following function element-wise.
```
Maybe:
```suggestion
    r"""Apply the SoftSign function element-wise.
```

svekars commented on torch/nn/functional.py:
```diff
@@ -1866,7 +1859,7 @@ def softmax(input: Tensor, dim: Optional[int] = None, _stacklevel: int = 3, dtyp
 def gumbel_softmax(logits: Tensor, tau: float = 1, hard: bool = False, eps: float = 1e-10, dim: int = -1) -> Tensor:
     r"""
-    Samples from the Gumbel-Softmax distribution (`Link 1`_  `Link 2`_) and optionally discretizes.
+    Sample from the Gumbel-Softmax distribution (`Link 1`_  `Link 2`_) and optionally discretizes.
```
```suggestion
    Sample from the Gumbel-Softmax distribution (`Link 1`_  `Link 2`_) and optionally discretize.
```

svekars commented on torch/nn/functional.py:
```diff
@@ -2133,7 +2129,7 @@ def embedding(
     sparse: bool = False,
 ) -> Tensor:
-    r"""A simple lookup table that looks up embeddings in a fixed dictionary and size.
+    r"""Lookup table that looks up embeddings in a fixed dictionary and size.
```
In the above phrase, it should be A lookup table (noun). It needs to be rephrased to start with a verb. Perhaps, "Generate a simple lookup table"?

svekars commented on torch/nn/functional.py:
```diff
@@ -2564,9 +2558,10 @@ def group_norm(
-    Applies normalization across channels.
+    r"""Apply local response normalization over an input signal.
+    
+    Input ignal is composed of several input planes, where channels occupy the second dimension.
```
```suggestion
    Input signal is composed of several input planes, where channels occupy the second dimension.
```

svekars commented on torch/nn/functional.py:
```diff
@@ -3066,8 +3063,7 @@ def binary_cross_entropy(
 ) -> Tensor:
-    r"""Function that measures the Binary Cross Entropy between the target and input
-    probabilities.
+    r"""Calculate Binary Cross Entropy between the target and input probabilities.
```
Probably better stick to the original "measure" rather than calculate:
```suggestion
    r"""Measure Binary Cross Entropy between the target and input probabilities.
```

svekars reviewed (commented):
I did a quick editorial pass. There are a few places where I'm not sure if rewrites a correct. Hopefully, @mikaylagawarecki or other reviewers can take a look.

pilot-j committed:
Update torch/nn/functional.py

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

pilot-j committed:
Update torch/nn/functional.py, line 1284

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

pilot-j committed:
Update torch/nn/functional.py, line 1286

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

pilot-j committed:
Update torch/nn/functional.py, line 1322

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

pilot-j committed:
Update torch/nn/functional.py, line 1325

Co-authored-by: Svetlana Karslioglu <svekars@meta.com>

pilot-j committed:
Update functional.py - after review

Changes -
1) Corrected bracket issue in dropout functions. Instead of removing the parentheses, I have included them at the right place. No brackets have been left unclosed. Reference - torch.nn docs

2) Minor corrections in response to reviews provided.

pilot-j commented on torch/nn/functional.py:
```diff
@@ -1966,9 +1959,9 @@ def log_softmax(input: Tensor, dim: Optional[int] = None, _stacklevel: int = 3,
-    r"""tanh(input) -> Tensor
-
-    Applies element-wise,
+    r"""Apply hyperbolic tangent (tanh) function element-wise.
```
Since this is the first line it should end with a period, I think. I have changed the summary portion.

pilot-j commented:
Accommodated relevant changes to address issues as highlighted in the first review.  
@svekars @mikaylagawarecki @jbschlosser kindly review. 
Let me know what other changes are necessary to ensure quality.
Thanks!!!

pilot-j committed:
removed docstring errors.

pilot-j committed:
fixed docstring errors

pilot-j committed:
fixed docstring errors

pilot-j committed:
Update _functional.py

pilot-j committed:
Update __init__.py

pilot-j committed:
Update lbfgs.py

pilot-j committed:
Update sparse_adam.py

pilot-j committed:
Update adadelta.py

pilot-j committed:
Update adagrad.py

pilot-j committed:
Update adam.py

pilot-j committed:
Update adamax.py

pilot-j committed:
Update adamw.py

pilot-j committed:
Update asgd.py

pilot-j committed:
Update random.py - reverting commit

pilot-j committed:
Update autocast_mode.py - reverting commits

pilot-j committed:
Update grad_scaler.py - reverting commits

pilot-j committed:
Update _functional.py - reverting commits

pilot-j committed:
Update __init__.py - reverting commits

pilot-j committed:
Update lbfgs.py - reverting commits

pilot-j committed:
Update asgd.py - reverting commits

pilot-j committed:
Update adamw.py - reverting changes

pilot-j committed:
Update adamax.py - reverting commits

pilot-j committed:
Update adam.py - reverting commits

pilot-j committed:
Update adagrad.py - reverting changes

pilot-j committed:
Update adadelta.py - reverting changes

pilot-j committed:
Update sparse_adam.py - reverting commits

mikaylagawarecki commented on torch/nn/functional.py:
```diff
@@ -2251,9 +2241,9 @@ def embedding_bag(
-    r"""Computes sums, means or maxes of `bags` of embeddings, without instantiating the
-    intermediate embeddings.
-
+    r"""Directly compute sums, means or maxes of `bags` of embeddings.
```
```suggestion
    r"""Compute sums, means or maxes of `bags` of embeddings.
```

mikaylagawarecki commented on torch/nn/functional.py:
```diff
@@ -2251,9 +2241,9 @@ def embedding_bag(
-
+    r"""Directly compute sums, means or maxes of `bags` of embeddings.
+    
+    Calculation done without instantiating the intermediate embeddings.
```
```suggestion
    Calculation is done without instantiating the intermediate embeddings.
```

mikaylagawarecki commented on torch/nn/functional.py:
```diff
@@ -2564,9 +2553,10 @@ def group_norm(
-    Applies normalization across channels.
+    r"""Apply local response normalization over an input signal.
+    
+    Input signal is composed of several input planes, where channels occupy the second dimension.
```
```suggestion
    The input signal is composed of several input planes, where channels occupy the second dimension.
```

mikaylagawarecki commented on torch/nn/functional.py:
```diff
@@ -2676,7 +2666,7 @@ def nll_loss(
     reduction: str = "mean",
 ) -> Tensor:
-    r"""The negative log likelihood loss.
+    r"""Apply the negative log likelihood loss.
```
For consistency, perhaps we can have the same word at the start of all the first sentences for the loss functions (e.g. perhaps "Compute"?)
```suggestion
    r"""Compute the negative log likelihood loss.
```

pilot-j replied:
Yes I have changed it accordingly.

mikaylagawarecki commented on torch/nn/functional.py:
```diff
@@ -2892,7 +2882,9 @@ def kl_div(
     log_target: bool = False,
 ) -> Tensor:
-    r"""The `Kullback-Leibler divergence Loss
+    r"""Calculate KL Divergence loss.
```
```suggestion
    r"""Compute the KL Divergence loss.
```

mikaylagawarecki commented on torch/nn/functional.py:
```diff
@@ -3066,8 +3058,7 @@ def binary_cross_entropy(
 ) -> Tensor:
-    r"""Function that measures the Binary Cross Entropy between the target and input
-    probabilities.
+    r"""Measure Binary Cross Entropy between the target and input probabilities.
```
```suggestion
    r"""Compute the Binary Cross Entropy loss between the target and input probabilities.
```

mikaylagawarecki commented on torch/nn/functional.py:
```diff
@@ -3136,8 +3127,7 @@ def binary_cross_entropy_with_logits(
 ) -> Tensor:
-    r"""Function that measures Binary Cross Entropy between target and input
-    logits.
+    r"""Caculate Binary Cross Entropy between target and input logits.
```
```suggestion
    r"""Compute the Binary Cross Entropy loss between target and input logits.
```

mikaylagawarecki replied:
```suggestion
    r"""Calculate Binary Cross Entropy between target and input logits.
```

mikaylagawarecki commented on torch/nn/functional.py:
```diff
@@ -3208,7 +3198,9 @@ def smooth_l1_loss(
     beta: float = 1.0,
 ) -> Tensor:
-    r"""Function that uses a squared term if the absolute
+    r"""Calculate Smooth L1 loss.
```
```suggestion
    r"""Compute the Smooth L1 loss.
```

mikaylagawarecki commented on torch/nn/functional.py:
```diff
@@ -3248,9 +3240,14 @@ def huber_loss(
     delta: float = 1.0,
 ) -> Tensor:
-    r"""Function that uses a squared term if the absolute
+    r"""Calculate Huber loss.
```
```suggestion
    r"""Compute the Huber loss.
```

mikaylagawarecki commented on torch/nn/functional.py:
```diff
@@ -3310,9 +3305,7 @@ def mse_loss(
-    r"""mse_loss(input, target, size_average=None, reduce=None, reduction='mean') -> Tensor
-
-    Measures the element-wise mean squared error.
+    r"""Measure the element-wise mean squared error.
```
```suggestion
    r"""Compute the element-wise mean squared error.
```

mikaylagawarecki commented on torch/nn/functional.py:
```diff
@@ -3379,7 +3369,11 @@ def hinge_embedding_loss(
     reduction: str = "mean",
 ) -> Tensor:
-    r"""hinge_embedding_loss(input, target, margin=1.0, size_average=None, reduce=None, reduction='mean') -> Tensor
+    r"""Measure the loss given an input tensor x and a labels tensor y (containing 1 or -1).
```
```suggestion
    r"""Compute the loss given an input tensor x and a labels tensor y (containing 1 or -1).
```

mikaylagawarecki commented on torch/nn/functional.py:
```diff
@@ -3343,10 +3336,7 @@ def margin_ranking_loss(
-
-    See :class:`~torch.nn.MarginRankingLoss` for details.
-    """
+    r"""See :class:`~torch.nn.MarginRankingLoss` for details."""
```
```suggestion
    r"""
    Compute the margin ranking loss.
   
    See :class:`~torch.nn.MarginRankingLoss` for details."""
```

mikaylagawarecki commented on torch/nn/functional.py:
```diff
@@ -3408,10 +3402,7 @@ def multilabel_margin_loss(
-
-    See :class:`~torch.nn.MultiLabelMarginLoss` for details.
-    """
+    r"""See :class:`~torch.nn.MultiLabelMarginLoss` for details."""
```
```suggestion
    r"""
    Compute the multi-label margin loss.
    
    See :class:`~torch.nn.MultiLabelMarginLoss` for details."""
```

mikaylagawarecki commented on torch/nn/functional.py:
```diff
@@ -3436,8 +3427,9 @@ def soft_margin_loss(
-    r"""soft_margin_loss(input, target, size_average=None, reduce=None, reduction='mean') -> Tensor
-
+    r"""
+    Optimize a two-class classification logistic loss between input tensor x and target tensor y (containing 1 or -1).
```
```suggestion
    Compute a two-class classification logistic loss between input tensor x and target tensor y (containing `1` or `-1`).
```

mikaylagawarecki commented on torch/nn/functional.py:
```diff
@@ -3459,7 +3451,9 @@ def multilabel_soft_margin_loss(
     reduction: str = "mean",
 ) -> Tensor:
-    r"""multilabel_soft_margin_loss(input, target, weight=None, size_average=None, reduce=None, reduction='mean') -> Tensor
+    r"""Optimize a multi-label one-versus-all loss based on max-entropy.
```
```suggestion
    r"""Compute a multi-label one-versus-all loss based on max-entropy.
```

mikaylagawarecki commented on torch/nn/functional.py:
```diff
@@ -3507,7 +3501,12 @@ def cosine_embedding_loss(
     reduction: str = "mean",
 ) -> Tensor:
-    r"""cosine_embedding_loss(input1, input2, target, margin=0, size_average=None, reduce=None, reduction='mean') -> Tensor
+    r"""Calculate Cosine Embedding loss.
```
```suggestion
    r"""Compute the Cosine Embedding loss.
```

mikaylagawarecki commented on torch/nn/functional.py:
```diff
@@ -4605,8 +4607,9 @@ def triplet_margin_loss(
 ) -> Tensor:
-    r"""
-    See :class:`~torch.nn.TripletMarginLoss` for details
+    r"""Measure triplet loss between given input tensors and a margin greater than 0.
```
```suggestion
    r"""Compute the triplet loss between given input tensors and a margin greater than 0.
```

mikaylagawarecki reviewed (commented):
Could you fix lint please

 running `make setup_lint` and `make quicklint` from your local pytorch directory and following the corresponding instructions should be sufficient, see [here](https://github.com/pytorch/pytorch/blob/4721553431e968ee5e90753346d92252cce98e84/Makefile#L25-L34)

mikaylagawarecki commented on torch/nn/functional.py:
```diff
@@ -5125,7 +5128,10 @@ def multi_head_attention_forward(
     is_causal: bool = False,
 ) -> Tuple[Tensor, Optional[Tensor]]:
-    r"""
+    r"""Foward method for MultiHeadAttention.
```
```suggestion
    r"""Forward method for MultiHeadAttention.
```

pilot-j committed:
Update functional.py - fixing lint issues

pilot-j committed:
Update functional.py - edited docstrings after review

pilot-j committed:
Update functional.py - minor correction

pilot-j commented on torch/nn/functional.py:
```diff
@@ -436,7 +436,7 @@ def fractional_max_pool2d_with_indices(
 ) -> Tuple[Tensor, Tensor]:
     r"""
-    fractional_max_pool2d(input, kernel_size, output_size=None, output_ratio=None, return_indices=False, _random_samples=None)
+    fractional_max_pool2d(input, kernel_size, output_size=None, output_ratio=None, return_indices=False, _random_samples=None).
```
I get your point @mikaylagawarecki. I guess the lint issues are now fixed, please let me know if I should remove the period or not.
Since the description line should end with a period, what can be the way around in these cases then?

mikaylagawarecki replied:
I think it is okay to remove the period and ignore the pydocstyle errors for these cases, thanks! :)

pilot-j replied:
@mikaylagawarecki Done!
Can you please review again and let me know is the PR fit for a merge or not? Let me know if there is need of any other fixes. :))

pilot-j committed:
Update functional.py - removed trailing spaces

pilot-j committed:
Update functional.py - removed periods from fn defn

svekars commented on torch/nn/functional.py:
```diff
@@ -1116,8 +1119,7 @@ def adaptive_max_pool2d_with_indices(
 ) -> Tuple[Tensor, Tensor]:
-    r"""
-    adaptive_max_pool2d(input, output_size, return_indices=False)
+    r"""adaptive_max_pool2d(input, output_size, return_indices=False)
```
you can put a comment like this next to it
```
# noqa D402
```
```suggestion
    r"""adaptive_max_pool2d(input, output_size, return_indices=False) # noqa D402
```

svekars commented on torch/nn/functional.py:
```diff
@@ -1116,8 +1119,7 @@ def adaptive_max_pool2d_with_indices(
 ) -> Tuple[Tensor, Tensor]:
-    r"""
-    adaptive_max_pool2d(input, output_size, return_indices=False)
+    r"""adaptive_max_pool2d(input, output_size, return_indices=False)
```
You can add a comment `# noqa D400`:
```suggestion
    r"""adaptive_max_pool2d(input, output_size, return_indices=False) # noqa D400
```

pilot-j replied:
I have added #noqa: D400 at appropriate places. Have run the code to check any errors (no docstring errors). Please review.
@svekars @mikaylagawarecki

svekars replied:
I don't think this: ` -> Tuple[Tensor, Tensor]: #noqa: D400` the correct way to apply that fix - you only apply to docstrings that have a function signature in them (like in the example above). pydocstyle errors because it wants a period at the end of that docstring. When you say # noqa D400 - you say "ignore the error that requires a period at the end of that sentence". You don't need to put it next to the missing docstrings errors - those errors as we said before can stay.

pilot-j replied:
I am a bit confused.
For clarification, I have not used noqa for missing docstring errors. I have used it only where we have function signatures that need to be there as pointed out by @mikaylagawarecki.
I did try to fix it by adding #noqa D400 inside docstring (as suggested above) - however, that did not resolve the D400 errors. On inspecting further, adding noqa after function ignores D400 for that function docstring. 
Current local output:
```
/content/pytorch/torch/nn/functional.py:3712 in public function `upsample` (skipping F811,B950):
        D103: Missing docstring in public function
/content/pytorch/torch/nn/functional.py:3717 in public function `upsample` (skipping F811,B950):
        D103: Missing docstring in public function
/content/pytorch/torch/nn/functional.py:3801 in public function `interpolate` (skipping F811,B950):
        D103: Missing docstring in public function
/content/pytorch/torch/nn/functional.py:3806 in public function `interpolate` (skipping F811,B950):
        D103: Missing docstring in public function
/content/pytorch/torch/nn/functional.py:3811 in public function `interpolate` (skipping F811,B950):
        D103: Missing docstring in public function
/content/pytorch/torch/nn/functional.py:3816 in public function `interpolate` (skipping F811):
        D103: Missing docstring in public function
/content/pytorch/torch/nn/functional.py:4071 in public function `upsample_nearest` (skipping F811):
        D103: Missing docstring in public function
/content/pytorch/torch/nn/functional.py:4076 in public function `upsample_nearest` (skipping F811):
        D103: Missing docstring in public function
/content/pytorch/torch/nn/functional.py:4109 in public function `upsample_bilinear` (skipping F811):
        D103: Missing docstring in public function
/content/pytorch/torch/nn/functional.py:4116 in public function `upsample_bilinear` (skipping F811):
        D103: Missing docstring in public function
/content/pytorch/torch/nn/functional.py:4123 in public function `upsample_bilinear` (skipping F811):
        D103: Missing docstring in public function
/content/pytorch/torch/nn/functional.py:4130 in public function `upsample_bilinear` (skipping F811):
        D103: Missing docstring in public function
/content/pytorch/torch/nn/functional.py:4745 in public function `assert_int_or_pair`:
        D103: Missing docstring in public function
13
```

Let me know if this is technically correct or not?

pilot-j committed:
Update functional.py - added noqa D400

kit1980 commented on torch/nn/functional.py:
```diff
@@ -3540,7 +3544,7 @@ def multi_margin_loss(
     reduce: Optional[bool] = None,
     reduction: str = "mean",
 ) -> Tensor:
-    r"""multi_margin_loss(input, target, p=1, margin=1, weight=None, size_average=None, reduce=None, reduction='mean') -> Tensor
```
Why do we want to remove the signature?

mikaylagawarecki replied:
I think for this case, this is ok, the preview still renders ok https://docs-preview.pytorch.org/pytorch/pytorch/112856/generated/torch.nn.functional.multi_margin_loss.html#torch.nn.functional.multi_margin_loss

pilot-j replied:
@mikaylagawarecki should I accomodate any changes to reflect this?
Do let me know. :)

pilot-j replied:
Ok I ll do it. 
I hope that should be all? 
Also @mikaylagawarecki can you please help me fix the two checks that are failing ?

pilot-j replied:
@mikaylagawarecki  Do I need function signatures of activation functions as well? (relu, glu, etc)

kit1980 commented:
@pilot-j please fix lint (at least two spaces before inline comment).

pilot-j commented:
> @pilot-j please fix lint (at least two spaces before inline comment).

@kit1980 Yeah. I am waiting to know if the #noqa: D400 after function is fine in this case or not. If it is ok, I can proceed with lint issues and will fix them asap. 
cc @svekars @mikaylagawarecki

svekars commented:
> > @pilot-j please fix lint (at least two spaces before inline comment).
> 
> @kit1980 Yeah. I am waiting to know if the #noqa: D400 after function is fine in this case or not. If it is ok, I can proceed with lint issues and will fix them asap. cc @svekars @mikaylagawarecki

Sorry for the confusion, your fixes were correct. Only lint errors left.

pilot-j committed:
Update functional.py - fixing lint

svekars commented:
@pytorchbot rebase

pilot-j committed:
Update functional.py - adding noqa D400,D402

pilot-j commented:
Please review recent changes. 
@mikaylagawarecki @svekars Can you please approve workflows, so that checks can run? 
This will help me to fix hidden errors if any (lint / build)
cc @kit1980

pilot-j commented:
@mikaylagawarecki @svekars  Can you please do a quick review and let me know do I need to change anything else or not? 
Sorry to disturb, I just want to make sure this PR is in a good shape.
cc @kit1980

kit1980 commented:
@pytorchbot merge
